sdk/storage/storage-blob/src/BlobServiceClient.ts

```diff
@@ -381,6 +382,9 @@ export class BlobServiceClient extends StorageClient {
       | Pipeline,
     options?: StoragePipelineOptions
   ) {
+    if (options && !options.httpClient) {
+      options.httpClient = new DefaultHttpClient();
```

Comment:
small nit: since the options object is passed in by the consumer, I prefer to not mutate it directly, e.g.

```ts
const newOptions = {
  httpClient: new DefaultHttpClient(),
  ...options
};
```

Reply:
This would create an instance of DefaultHttpClient first, then possibly be overridden by options.httpClient?

Reply:
Yeah, another way to do it is still put an if statement around it so you can avoid making an empty default client if one exists